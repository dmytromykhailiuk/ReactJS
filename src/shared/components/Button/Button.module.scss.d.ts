// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'button': string;
  'button--big': string;
  'button--primary': string;
  'button--secondary': string;
  'button--small': string;
}
export const cssExports: CssExports;
export default cssExports;
