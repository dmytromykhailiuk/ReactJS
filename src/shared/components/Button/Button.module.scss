@import "scss/variables";

.button {
  letter-spacing: 1.5px;
  cursor: pointer;
  outline: none;
  box-shadow: 0 0 1px $dark-grey-color;
  text-shadow: 0 0 1px $dark-grey-color;
  border-radius: 5px;
  transition: 0.3s;

  &--primary {
    background-color: $pink-color;
    border: solid 2px $pink-color;
    color: $white-color;

    &:hover {
      background-color: transparent;
      color: $pink-color;
    }
  }

  &--secondary { 
    color: $pink-color;
    background-color: transparent;
    border: solid 2px $pink-color;
  
    &:hover {
      color: $white-color;
      border-color: $white-color;
    }
  }

  &--big {
    height: 66px;
    width: 240px;
    font-size: 24px;
  }

  &--small {
    height: 48px;
    width: 191px;
    font-size: 20px;
    background-color: rgba($light-grey-color, 0.8);
    text-shadow: none;
    border: none;
    box-shadow: none;

    &:hover {
      background-color: rgba($light-grey-color, 0.4);
    }
  }
}
