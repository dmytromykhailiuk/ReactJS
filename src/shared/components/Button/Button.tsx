import React, { SyntheticEvent } from 'react';
import { ButtonTypes, ButtonSize } from "../../enums";
import classes from "./Button.module.scss";
import classnames from "classnames";

interface ButtonProps {
  type?: ButtonTypes;
  isSubmit?: boolean;
  size?: ButtonSize;
  disabled?: boolean;
  onButtonClicked?: (event: SyntheticEvent) => void;
}

const Button:React.FC<ButtonProps> = ({
  children, type = ButtonTypes.PRIMARY, size = ButtonSize.BIG, onButtonClicked = () => {}, isSubmit = false, disabled = false 
}) => {
  return (
    <button 
      disabled={disabled}
      type={isSubmit ? "submit" : "button"}
      className={`
        ${classes.button} 
        ${classnames({[classes["button--disabled"]]: disabled})} 
        ${classes[type === ButtonTypes.PRIMARY ? "button--primary": "button--secondary"]} 
        ${classes[size === ButtonSize.BIG ? "button--big": "button--small"]} 
      `}
      onClick={onButtonClicked}
    >
      {children}
    </button>
  )
}

export default Button;
