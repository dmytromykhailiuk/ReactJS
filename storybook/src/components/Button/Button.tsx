import React, { SyntheticEvent } from 'react';
import classnames from 'classnames';
import { ButtonTypes, ButtonSize } from '../../enums';
import classes from './Button.module.scss';

export interface ButtonProps {
  type?: ButtonTypes;
  isSubmit?: boolean;
  size?: ButtonSize;
  disabled?: boolean;
  style?: React.CSSProperties;
  onButtonClicked?: (event: SyntheticEvent) => void;
}

const Button: React.FC<ButtonProps> = ({
  children,
  type = ButtonTypes.PRIMARY,
  size = ButtonSize.BIG,
  onButtonClicked = () => {},
  isSubmit = false,
  disabled = false,
  style = {},
}) => (
  <button
    disabled={disabled}
    type={isSubmit ? 'submit' : 'button'}
    className={`
        ${classes.button} 
        ${classnames({ [classes['button--disabled']]: disabled })} 
        ${classes[type === ButtonTypes.PRIMARY ? 'button--primary' : 'button--secondary']} 
        ${classes[size === ButtonSize.BIG ? 'button--big' : 'button--small']} 
      `}
    style={style}
    onClick={onButtonClicked}
  >
    {children}
  </button>
);

export default Button;
