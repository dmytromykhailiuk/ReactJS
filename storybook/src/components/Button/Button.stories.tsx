import React from 'react';
import { Story, Meta } from '@storybook/react';

import Button, { ButtonProps } from './Button';
import { ButtonTypes } from '../../enums/button-types';

export default {
  title: 'Example/Button',
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (args: any) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  children: "Button",
  type: ButtonTypes.PRIMARY,
};

export const Secondary = Template.bind({});
Secondary.args = {
  children: "Button",
  type: ButtonTypes.SECONDARY,
};

