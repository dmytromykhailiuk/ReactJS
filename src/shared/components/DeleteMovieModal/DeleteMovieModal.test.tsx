import { shallow } from "enzyme";
import React from "react";
import { DeleteMovieModal } from "./";
import toJson from 'enzyme-to-json';
import { Button } from '../Button';

describe("DeleteMovieModal", () => {
  it("should match first snepshot", () => {
    const wrapper = shallow(
      <DeleteMovieModal
        movie={{} as any}
        onCloseModal={() => {}}
        onCloseWithSaving={() => {}}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot("first");
  })

  it("should call onCloseWithSaving with movie when click on button occurred", () => {
    const onClick = jest.fn();
    const movie = {} as any;
    const wrapper = shallow(
      <DeleteMovieModal
        movie={movie}
        onCloseModal={() => {}}
        onCloseWithSaving={onClick}
      />
    )
    wrapper.find(Button).dive().find("button").simulate("click");
    expect(onClick).toHaveBeenCalledWith(movie);
  })
})
