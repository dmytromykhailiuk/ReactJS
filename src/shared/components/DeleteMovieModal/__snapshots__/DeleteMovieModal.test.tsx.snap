// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovieModal should match first snepshot: first 1`] = `
<Component
  header="DELETE MOVIE"
  onCloseModal={[Function]}
>
  <div
    className="delete-movie-modal__body"
  >
    Are you sure you want to delete this movie?
  </div>
  <div
    className="delete-movie-modal__control"
  >
    <Button
      onButtonClicked={[Function]}
    >
      Confirm
    </Button>
  </div>
</Component>
`;
